fabric-fluid-extensions-v1/src/main/java/net/fabricmc/fabric/api/fluid/extension/v1/FabricFlowableFluid.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.fluid.extension.v1;
+
+import net.fabricmc.fabric.impl.fluid.extension.FlowableFluidExtensions;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.fluid.FlowableFluid;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.state.StateManager;
+import net.minecraft.state.property.IntProperty;
+
+public abstract class FabricFlowableFluid extends FlowableFluid {
+	private IntProperty stateIndexProperty;
+	
+	public FabricFlowableFluid() {
+		((FlowableFluidExtensions) this).setMaxLevel(getLevels());
+		setDefaultState(getDefaultState().with(stateIndexProperty, 0));
+	}
+
+	@Override
+	public Fluid getStill() {
+		return this;
+	}
+	
+	@Override
+	public Fluid getFlowing() {
+		return this;
+	}
+
+	@Override
+	public FluidState getStill(boolean falling) {
+		if (falling) {
+			return getStill().getDefaultState().with(stateIndexProperty, ((FlowableFluidExtensions) this).getMaxLevel());
+		} else {
+			return getStill().getDefaultState();
```

Comment:
you can drop the redundant else here and just return at the end of the method the default state.